internal/controller/infrastructure/remote_machine_controller.go

```diff
@@ -81,6 +85,14 @@ func (r *RemoteMachineController) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	if rm.Spec.Pool != "" {
+		err := r.findPooledMachine(ctx, rm)
+		if err != nil {
+			log.Error(err, "Error reserving PooledMachine")
+			return ctrl.Result{Requeue: true}, err
+		}
+	}
+
```

Comment:
Note for myself: add validation for the RemoteMachine.spec fields since all of them are optional now

Reply:
✅